mutation_compactor.hh

```diff
@@ -292,11 +297,14 @@ public:
         , _last_dk({dht::token(), partition_key::make_empty()})
         , _last_pos(position_in_partition::end_of_partition_tag_t())
         , _collector(std::make_unique<mutation_compactor_garbage_collector>(_schema))
+        , _validator("mutation_compactor for compaction", _schema, mutation_fragment_stream_validation_level::partition_region)
     {
         static_assert(sstable_compaction(), "This constructor can only be used for sstable compaction.");
     }
 
     void consume_new_partition(const dht::decorated_key& dk) {
+        _validator(mutation_fragment_v2::kind::partition_start, position_in_partition_view(position_in_partition_view::partition_start_tag_t{}), {});
```

Comment:
We should add `position_in_partition_view::for_partition_start()`.

Reply:
Yes.